### PR TITLE
Display correct message when editing distance requests

### PR DESCRIPTION
**Pull request by Gonals**
### Fixed Issues

$ https://github.com/Expensify/App/issues/27630
PROPOSAL: 

### Tests
- Create a distance request in Newdot following the steps [here](https://github.com/Expensify/Web-Expensify/pull/38436)
- Once it shows up in newdot, open it and click on `Distance`.
- Edit the waypoints and wait for a few seconds until it is done loading.
- Confirm the following message displays:
Screenshot 2023-09-21 at 5 22 07 PM

- [x] Verify that no errors appear in the JS console

### Offline tests
None

### QA Steps
- Create a distance request in Newdot by opening a chat with a Workspace and clicking + > Request Money > Distance and setting waypoints.
- Open it and click on `Distance`.
- Edit the waypoints and wait for a few seconds until it is done loading.
- Confirm the following message displays:
Screenshot 2023-09-21 at 5 22 07 PM

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-09-22 at 11 55 02 AM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-09-22 at 12 50 27 PM

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-09-22 at 12 47 27 PM

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-09-22 at 11 57 20 AM

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

Screenshot 2023-09-22 at 12 26 46 PM

</details>

### CODE REVIEW

File: src/libs/actions/IOU.js [hunk 1]
```diff
@@ -630,6 +630,175 @@ function createDistanceRequest(report, participant, comment, created, transactio
...
+        });
+        modifiedReportActionFailureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThread.reportID}`,
+            value: {
+                [updatedReportAction.reportActionID]: updatedReportAction,
+            },
+        });
+
+        // Step 4: Comput the IOU total and update the report preview message (and report header) so LHN amount owed is correct.
```
Comment (cubuspl42) [inline src/libs/actions/IOU.js:701]: Typo?

File: src/pages/iou/MoneyRequestEditWaypointPage.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            waypointIndex: PropTypes.string,
+        }),
+    }),
+};
+
+const defaultProps = {
+    route: {},
+};
+
+// This component is responsible for grabbing the transactionID from the IOU key
```
Comment (cubuspl42) [inline src/pages/iou/MoneyRequestEditWaypointPage.js:28]: Is it some magic I'm not familiar with? This component seems to do nothing beyond forwarding the `route` property

Comment (cubuspl42) [reply src/pages/iou/MoneyRequestEditWaypointPage.js:28]: Or was this comment copy-pasted?

**Comment by cubuspl42** [comment]
Well, it's a big one. To be honest, I'm not completely sure I'm experienced enough to review the high-level decisions here. I left minor comments and can help with the testing.

**Comment by Gonals** [comment]
> Well, it's a big one. To be honest, I'm not completely sure I'm experienced enough to review the high-level decisions here. I left minor comments and can help with the testing.

Ah! Don't worry too much about it. It looks big because it comes from https://github.com/Expensify/App/pull/26141 's branch. Once that one merges, it will become simple :)

**Comment by Gonals** [comment]
@cubuspl42 Ready for review! Should be waaaay simpler now 😁

File: src/libs/ReportUtils.js [hunk 3]
```diff
@@ -1565,6 +1565,28 @@ function getProperSchemaForModifiedExpenseMessage(newValue, oldValue, valueName,
...
 
+/**
+ * Get the proper message schema for modified expense message.
+ *
+ * @param {String} newValue
+ * @param {String} oldValue
+ * @param {String} valueName
+ * @param {Boolean} valueInQuotes
+ * @returns {String}
+ */
```
Comment (neil-marcellini) [inline src/libs/ReportUtils.js:1576]: Looks like this was pasted from above but you forgot to update it

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
Looks great, assuming the C+ testing goes well! There's one JSDoc that needs to be updated.

**Comment by cubuspl42** [comment]
> Create a distance request in Newdot following the steps [here](https://github.com/Expensify/Web-Expensify
> Confirm the following message displays:

> Looks great, assuming the C+ testing goes well! There's one JSDoc that needs to be updated.

I don't have access to the steps nor the screenshot... Maybe they can be copied here?

File: src/libs/ReportUtils.js (hunk 3 above)
Comment (Gonals) [reply src/libs/ReportUtils.js]: Updated!

**Comment by Gonals** [comment]
> > Create a distance request in Newdot following the steps [here](https://github.com/Expensify/Web-Expensify
> > Confirm the following message displays:
> 
> > Looks great, assuming the C+ testing goes well! There's one JSDoc that needs to be updated.
> 
> I don't have access to the steps nor the screenshot... Maybe they can be copied here?

Huh. Weird.

Here are the test steps:

- Create a distance request in Newdot following the steps [here](https://github.com/Expensify/Web-Expensify/pull/38436). You can also try simply clicking (in a workspace chate) : `+` > `Request Money` > `Distance` and see if the map loads for you 
- Once it shows up in newdot, open it and click on `Distance`.
- Edit the waypoints and wait for a few seconds until it is done loading.
- Confirm the following message displays:
Screenshot 2023-09-21 at 5 22 07 PM

**Comment by cubuspl42** [comment]
@Gonals I just don't have access to https://github.com/Expensify/Web-Expensify, but that's typical for C+ team, I think. The screenshot still doesn't work, but maybe I'll figure things out without it

**Comment by Gonals** [comment]
> I just don't have access to https://github.com/Expensify/Web-Expensify

Ah, gotcha! Yeah, try creating it from newDot doing `+ > Request Money > Distance` and see if the map loads for you (it takes a few seconds). Otherwise, this is all yours, @neil-marcellini.

The message should say: `changed the distance to XXXX (previously YYYY), which updated the amount to ZZZZ (previously MMMM)`

**Comment by cubuspl42** [comment]
@Gonals Would you please merge `main` and tell me if you are able to execute the testing steps?

I'm having trouble going past pressing "+" (I always test with latest `main` merged-in)

**Review by neil-marcellini — APPROVED** [review]
Looks good and tests well on web at least. In the interest of urgency I'm going to ask for another C+ in Slack to complete the checklist.

https://github.com/Expensify/App/assets/26260477/23535ee9-45d5-4012-819a-33c59b4ca4de

**Review by rushatgabhane — APPROVED** [review]
@neil-marcellini LGTM!

**Comment by JmillsExpensify** [comment]
Can I have an Expensify employee provide a payment summary so that I can get @rushatgabhane paid?

**Comment by neil-marcellini** [comment]
Payment summary: $500 to @rushatgabhane for the PR review.

**Comment by JmillsExpensify** [comment]
$500 payment approved for @rushatgabhane based on summary above.

**Pull request merged**